python/fic_decode_vc.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# 
+# Copyright 2017 Moritz Luca Schmid, Communications Engineering Lab (CEL) / Karlsruhe Institute of Technology (KIT).
+# 
+# This is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+# 
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this software; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+# 
+
+from gnuradio import gr, trellis, blocks
+import dab
+from math import sqrt
+
+class fic_decode_vc(gr.hier_block2):
+    """
+    @brief block to decode FIBs (fast information blocks) from the FIC (fast information channel) of a demodulated DAB signal
+
+    - get demodulated FIC OFDM symbols from transmission frame
+    - do convolutional decoding
+    - undo energy dispersal
+    - get FIC information
+    """
+    def __init__(self, dab_params):
+        gr.hier_block2.__init__(self,
+            "fic_decode_vc",
+            gr.io_signature(1, 1, gr.sizeof_gr_complex * dab_params.num_carriers),  # Input signature
+            gr.io_signature(0, 0, gr.sizeof_char)) # Output signature
+
+        self.dp = dab_params
+
+        # complex to intereaved float for convolutional decoding
+        self.c2f = dab.complex_to_interleaved_float_vcf_make(self.dp.num_carriers)
+
+        # FIB block partitioning
+        self.v2s = blocks.vector_to_stream_make(gr.sizeof_float, self.dp.num_carriers * 2)
+        self.s2v = blocks.stream_to_vector_make(gr.sizeof_float, self.dp.fic_punctured_codeword_length)
+
+        # unpuncturing
+        self.unpuncture = dab.unpuncture_vff(self.dp.assembled_fic_puncturing_sequence, 0)
+
+        # convolutional coding
+        # self.fsm = trellis.fsm(self.dp.conv_code_in_bits, self.dp.conv_code_out_bits, self.dp.conv_code_generator_polynomials)
```

Comment:
remove unused code, especially if it might confuse the reader. Keep if containing helpful example e.g. for future modifications.